infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/query/parsing/impl/InfinispanRemoteBasedQueryParserService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.query.parsing.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+import java.util.Map;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.hql.QueryParser;
+import org.hibernate.hql.ast.spi.EntityNamesResolver;
+import org.hibernate.ogm.query.spi.BaseQueryParserService;
+import org.hibernate.ogm.query.spi.QueryParserService;
+import org.hibernate.ogm.query.spi.QueryParsingResult;
+import org.hibernate.ogm.service.impl.SessionFactoryEntityNamesResolver;
+import org.hibernate.ogm.util.impl.Log;
+import org.hibernate.ogm.util.impl.LoggerFactory;
+
+/**
+ * A {@link QueryParserService} implementation which creates Infinispan server queries in form of {@link String}s.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class InfinispanRemoteBasedQueryParserService extends BaseQueryParserService {
+
+	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
+
+	private volatile SessionFactoryEntityNamesResolver entityNamesResolver;
+
+	@Override
+	public QueryParsingResult parseQuery(SessionFactoryImplementor sessionFactory, String queryString, Map<String, Object> namedParameters) {
+		throw new UnsupportedOperationException( "The Infinispan Remote query parser supports parameterized queries" );
+	}
+
+	@Override
+	public QueryParsingResult parseQuery(SessionFactoryImplementor sessionFactory, String queryString) {
+		QueryParser queryParser = new QueryParser();
+		InfinispanRemoteProcessingChain processingChain = createProcessingChain( sessionFactory );
+		InfinispanRemoteQueryParsingResult result = queryParser.parseQuery( queryString, processingChain );
+
+		log.createdQuery( queryString, result );
```

Comment:
Is it going to be any real difference between queryString and result? Wouldn't be easier to return the queryString?

Reply:
For instance "SELECT p FROM org.java.package.Person p where p.username = 'Fabio'" became "FROM HibernateOGMGenerated.Person where user = 'Fabio'. Where the Entity field "username" is mapped as "user" as datastore column name.

Reply:
Thanks, that's what I wanted to know. Adding the tests will make this clearer.